tests/phpunit/Logger/GitHubAnnotationsLoggerTest.php

```diff
@@ -71,4 +93,44 @@ public static function metricsProvider(): iterable
             ],
         ];
     }
+
+    public function test_it_logs_correctly_with_ci_project_dir(): void
+    {
+        \Safe\putenv('GITHUB_WORKSPACE=/my/project/dir');
+
+        $resultsCollector = new ResultsCollector();
+        $resultsCollector->collect(
+            new MutantExecutionResult(
+                'bin/phpunit --configuration infection-tmp-phpunit.xml --filter "tests/Acme/FooTest.php"',
+                'process output',
+                DetectionStatus::ESCAPED,
+                now(normalize_trailing_spaces(
+                    <<<DIFF
+                        --- Original
+                        +++ New
+                        @@ @@
+
+                        - echo 'original';
+                        + echo 'escaped#0';
+
+                        DIFF,
+                )),
+                'a1b2c3',
+                MutatorName::getName(For_::class),
+                '/my/project/dir/foo/bar',
```

Comment:
If you extend `CreateMetricsCalculator` to include a static variable for a path prefix, you can avoid adding your own ResultsCollector.

The test will look like this:
```php
    public function test_it_logs_correctly_using_github_workspace(): void {
        \Safe\putenv('GITHUB_WORKSPACE=/my/project/dir');

        self::setOriginalFilePrefix('/my/project/dir/');

        $resultsCollector = self::createCompleteResultsCollector();

        $logger = new GitHubAnnotationsLogger($resultsCollector);

        $this->assertStringContainsString('warning file=foo/bar', $logger->getLogLines()[0]);
    }
```

Reply:
Yes, much better (Y)